### PR TITLE
Scene controller

### PR DESCRIPTION
**Pull request by taangle**
Added scene controller with simple methods to change scenes. Instance in each level would need to have nextLevel and previousLevel set manually. Could easily be changed to support different types of level loading.

### CODE REVIEW

File: Assets/Resources/Scripts/SceneController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		}
+	}
+
+	/// <summary>
+	/// Loads scene with specified name. Scene must be in build settings.
+	/// </summary>
+	/// <param name="targetSceneName"></param>
+	public static void GoToScene(string targetSceneName)
+	{
+		// Should this funtion be made a coroutine so that the caller is forced to be aware of the asynchronicity?
```
Comment (Runi-c) [inline Assets/Resources/Scripts/SceneController.cs]: If this function is made to return the result of LoadSceneAsync (an AsyncOperation) the caller can handle it any way it wants, which should be good enough for now

File: Assets/Resources/Scripts/SceneController.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		{
+			print("~~Scene path: " + scene.path);
+		}
+	}
+
+	/// <summary>
+	/// Loads scene with specified name. Scene must be in build settings.
+	/// </summary>
+	/// <param name="targetSceneName"></param>
+	public static void GoToScene(string targetSceneName)
```
Comment (Runi-c) [inline Assets/Resources/Scripts/SceneController.cs]: Shouldn't this function also update currentScene and previousScene (and probably nextScene as well)?

File: Assets/Resources/Scripts/SceneController.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+﻿using UnityEditor;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class SceneController : MonoBehaviour
+{
+	// currentScene populates in Start, previous and next need to be set manually
```
Comment (Runi-c) [inline Assets/Resources/Scripts/SceneController.cs]: the SceneController should probably be the one responsible for deciding which scene to go to next at any given step, for starters an array that the SceneController (which would be on a DontDestroyOnLoad object) steps forward and back through

**Review by Runi-c — CHANGES_REQUESTED** [review]
I cleaned up the .gitignore and made a few comments on the SceneController about things I was unclear about or think should be changed. Feel free to look over, make changes, and explain anything I missed

**Comment by taangle** [comment]
@Keterr I changed everything to fit what we talked about. SceneController is now a DontDestroyOnLoad, although I purposely didn't make it an actual "singleton" (although it functionally will be one). There's now a SceneChanger script that can be attached to any object wanting to use the SceneController.

**Review by Runi-c — APPROVED** [review]

**Pull request merged**